### ISSUE TITLE
Problem non-latin symbols

### ISSUE THREAD

**Issue by Zerstoren** [haraka/Haraka#2548]
### system info

Haraka | Haraka.js — Version: 2.8.21
 --- | :--- 
Node | v7.10.1
OS | Linux hostname 4.4.0-83-generic #106-Ubuntu SMP Mon Jun 26 17:54:43 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
openssl | OpenSSL 1.0.2g  1 Mar 2016

### Expected behavior
Content Transfer Encoding 8bit lose non-ASCII symbols

### Observed behavior
Emails from many of our clients containing Content-Type: utf-8 and Content-Transfer-Encoding 8bit removes non-ASCII characters. For example, German ß turns into �.

### Steps to reproduce
Create a plugin from this code

```javascript
var plugin;

exports.register = function () {
    plugin = this;
};

exports.hook_rcpt_ok = function (next, connection, params) {
    connection.transaction.parse_body = true;
    return next();
};

exports.hook_data_post = function (next, connection, params) {
    connection.logwarn(connection.transaction.body);
    return next();
};
```

and send this email

```
Subject: Test
To: to@example.com
From: from@example.com
Date: Wed, 27 Nov 2018 18:09:23 +0100
MIME-version: 1.0
Content-type: multipart/alternative;
 boundary=------------D0A00162984CC178E2583417
Content-language: en-US

This is a multi-part message in MIME format.
--------------D0A00162984CC178E2583417
Content-Type: text/plain; charset=utf-8; format=flowed
Content-Transfer-Encoding: 8bit

Grüße,

Buß

--------------D0A00162984CC178E2583417
Content-Type: text/html; charset=utf-8
Content-Transfer-Encoding: 8bit

<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  </head>
  <body text="#000000" bgcolor="#FFFFFF">
    <p>Grüße,

Buß<br>
  </body>
</html>

--------------D0A00162984CC178E2583417--
```
via swaks with command
```swaks --to to@example.com --from from@example.com -d test.eml```

As a result we see the following problem
```
       bodytext: 'Gr��e,\n\n Bu�\n\n',
       body_text_encoded: 'Grüße,\n\nBuß\n\n',
       body_encoding: 'utf-8',
```

It is not a problem with console. This will be displayed in DB and other places.

We were trying to find out what the problem was and found 

> 367 to 369 in [master](/haraka/Haraka/blob/master/mailbody.js)

`new Buffer(line, 'binary');` will lose all non ascii symbols, when 'binary' is removed everything is ok.

**Comment by msimerson** [comment]
This is tangentially related to #2374

**Comment by Zerstoren** [comment]
No, change from new Buffer() to Buffer.from not fix this problem. I try to do that.
Problem with encoding binary parametr.  By default Buffer work in 'utf8', but with 'binary' buffer work with latin1 encoding and lose all non latin symbols.

**Comment by smfreegard** [comment]
There's something weird going on here.   "binary" should work just fine with any character set encoding which is why it was used in the first place.   mailbody.js then converts it from binary to the correct encoding based on that the MIME headers tell it to.
